tf2_ros/include/tf2_ros/message_filter.h

```diff
@@ -555,142 +507,99 @@ class MessageFilter : public MessageFilterBase, public message_filters::SimpleFi
   /**
    * \brief Callback that happens when we receive a message on the message topic
    */
-  void incomingMessage(const ros::MessageEvent<M const>& evt)
+  void incomingMessage(const message_filters::MessageEvent<M const> & evt)
   {
     add(evt);
   }
 
   void checkFailures()
   {
-    if (next_failure_warning_.isZero())
-    {
-      next_failure_warning_ = ros::WallTime::now() + ros::WallDuration(15);
+    if (!next_failure_warning_.nanoseconds()) {
+      next_failure_warning_ = rclcpp::Clock::now() + rclcpp::Duration(15, 0);
     }
 
-    if (ros::WallTime::now() >= next_failure_warning_)
-    {
-      if (incoming_message_count_ - message_count_ == 0)
-      {
+    if (rclcpp::Clock::now() >= next_failure_warning_) {
+      if (incoming_message_count_ - message_count_ == 0) {
         return;
       }
 
-      double dropped_pct = (double)dropped_message_count_ / (double)(incoming_message_count_ - message_count_);
-      if (dropped_pct > 0.95)
-      {
-        TF2_ROS_MESSAGEFILTER_WARN("Dropped %.2f%% of messages so far. Please turn the [%s.message_notifier] rosconsole logger to DEBUG for more information.", dropped_pct*100, ROSCONSOLE_DEFAULT_NAME);
-        next_failure_warning_ = ros::WallTime::now() + ros::WallDuration(60);
-
-        if ((double)failed_out_the_back_count_ / (double)dropped_message_count_ > 0.5)
-        {
-          TF2_ROS_MESSAGEFILTER_WARN("  The majority of dropped messages were due to messages growing older than the TF cache time.  The last message's timestamp was: %f, and the last frame_id was: %s", last_out_the_back_stamp_.toSec(), last_out_the_back_frame_.c_str());
+      double dropped_pct = (double)dropped_message_count_ /
+        (double)(incoming_message_count_ - message_count_);
+      if (dropped_pct > 0.95) {
+        TF2_ROS_MESSAGEFILTER_WARN(
+          "Dropped %.2f%% of messages so far. Please turn the [%s.message_notifier] rosconsole logger to DEBUG for more information.", dropped_pct * 100,
+          "tf2_ros_message_filter");
+        next_failure_warning_ = rclcpp::Clock::now() + rclcpp::Duration(60, 0);
+
+        if ((double)failed_out_the_back_count_ / (double)dropped_message_count_ > 0.5) {
+          TF2_ROS_MESSAGEFILTER_WARN(
+            "  The majority of dropped messages were due to messages growing older than the TF cache time.  The last message's timestamp was: %f, and the last frame_id was: %s",
+            last_out_the_back_stamp_.seconds(), last_out_the_back_frame_.c_str());
         }
       }
     }
   }
 
-  struct CBQueueCallback : public ros::CallbackInterface
+  void messageDropped(const MEvent & evt, FilterFailureReason reason)
   {
-    CBQueueCallback(MessageFilter* filter, const MEvent& event, bool success, FilterFailureReason reason)
```

Comment:
I see you have dropped the `callback_queue` completely.  What's the reasoning for that?

Reply:
@gaoethan can you respond to this?

Reply:
OK, that makes sense.  Given that this is something we eventually want, I think it makes sense to comment this out instead of deleting this code.  I'm going to do that in a separate commit.